src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -0,0 +1,612 @@
+            }
+        }
+
+        /// <inheritdoc />
+        public IntPtr Handle
+        {
+            get
+            {
+                unsafe {
+                    return new IntPtr(WindowPtr);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public bool UseSingleThreadedWindow { get; }
+
+        private Point _position;
+
+        /// <inheritdoc />
+        public Point Position
+        {
+            get => _position;
+            set
+            {
+                glfwThread.Invoke(() =>
+                {
+                    unsafe {
+                        glfw.SetWindowPos(WindowPtr, value.X, value.Y);
+                    }
+                });
+
+                _position = value;
+            }
+        }
+
+        private Size _size;
+
+        /// <inheritdoc />
+        public Size Size
+        {
+            get => _size;
+            set
+            {
+                glfwThread.Invoke(() =>
+                {
+                    unsafe {
+                        glfw.SetWindowSize(WindowPtr, value.Width, value.Height);
+                    }
+                });
+
+                _size = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public double FramesPerSecond { get; }
+
+        /// <inheritdoc />
+        public double UpdatesPerSecond { get; }
+
+        /// <inheritdoc />
+        public GraphicsAPI API { get; }
+
+        private string _title;
+
+        /// <inheritdoc />
+        public string Title
+        {
+            get => _title;
+            set
+            {
+                glfwThread.Invoke(() =>
+                {
+                    unsafe {
+                        glfw.SetWindowTitle(WindowPtr, value);
+                    }
+                });
+
+                _title = value;
+            }
+        }
+
+        private WindowState _windowState;
+
+        /// <inheritdoc />
+        public WindowState WindowState
+        {
+            get => _windowState;
+            set
+            {
+                glfwThread.Invoke(() =>
+                {
+                    unsafe {
+                        switch (value) {
+                            case WindowState.Normal:
+                                glfw.RestoreWindow(WindowPtr);
+                                break;
+                            case WindowState.Minimized:
+                                glfw.IconifyWindow(WindowPtr);
+                                break;
+                            case WindowState.Maximized:
+                                glfw.MaximizeWindow(WindowPtr);
+                                break;
+                            case WindowState.Fullscreen:
+                                var monitor = glfw.GetWindowMonitor(WindowPtr);
+                                var mode = glfw.GetVideoMode(monitor);
+                                glfw.SetWindowMonitor(WindowPtr, monitor, 0, 0, mode->Width, mode->Height,
```

Comment:
Discussion point for a future ADM: We should design a monitor API to be implemented in a future PR, as I'd really like to get some sort of monitor API in, even if minimal. Not in scope for this PR though.